ca/config.go

```diff
@@ -112,6 +112,14 @@ func (s *SecurityConfig) RootCA() *RootCA {
 	return s.rootCA
 }
 
+// ExternalCA returns the external CA.
+func (s *SecurityConfig) ExternalCA() *ExternalCA {
+	s.mu.Lock()
+	defer s.mu.Unlock()
```

Comment:
I don't believe it's necessary to hold this lock. `s.externalCA` seems to be set when the `SecurityConfig` is created, and never changed to point to anything else. `updateCluster` uses this field without holding the lock.